modules/programs/thunderbird.nix

```
@@ -415,14 +432,24 @@ in {
           a.thunderbird.profiles == [ ]
           || any (p: p == name) a.thunderbird.profiles) enabledAccountsWithId;
 
+        orderedAccountsIds = let
+          accountsOrderIds =
+            map (a: builtins.hashString "sha256" a) profile.accountsOrder;
+          enabledAccountsIds = map (a: a.id) accounts;
+        in (lists.intersectLists enabledAccountsIds accountsOrderIds)
+        ++ (lists.subtractLists accountsOrderIds enabledAccountsIds);
+
```

This silently discards accounts in the order that are missing. Such accounts should either throw an error, or (to enable the use of multiple imperative accounts as noted above) be included verbatim.

---

I agree and I can do either depending on what is better. I would tend towards the second option.